data-generation/src/main/scala/SnapshotDelta.scala

```diff
@@ -75,55 +75,90 @@ object SnapshotDelta {
+    Graph(vertices, edges)
+  } )
+}
+def applyEdgeLogs(snapshot: Graph[LTSV.Attributes, LTSV.Attributes], logs: RDD[LogTSV]): RDD[Edge[Attributes]] = {
+  val previousSnapshotEdges = snapshot.edges.map(edge => ((edge.dstId, edge.srcId), edge.attr))
+
+  // Get all ids of edges in the log interval
+  val edgeIds = logs.flatMap(getEdgeIds)
+
+  // Squash all actions on each edge to one single action
+  val edgesWithAction = edgeIds.map(id => (id, logs.filter(log => log.objectType match {
+    case VERTEX(objId) => id._1 == objId || id._2 == objId
+    case EDGE(srcId, dstId) => id._1 == srcId && id._2 == dstId
+  })))
+    .map(edgeWithAction => (edgeWithAction._1, mergeLogTSVs(edgeWithAction._2)))
+
+  // Combine edges from the snapshot with the current log interval
+  val joinedEdges = previousSnapshotEdges.fullOuterJoin(edgesWithAction)
+
+  // Create, update or omit (delete) edges based on values are present or not.
+  val newSnapshotEdges = joinedEdges.flatMap(x => x._2 match {
+    case (Some(snapshotEdge), None) => Some(Edge(x._1._1, x._1._2, snapshotEdge))
+    case (Some(snapshotEdge), Some(newEdge)) => newEdge match {
+      case newEdge.action == DELETE => None
+      case newEdge.action == UPDATE => Some(Edge(x._1._1, x._1._2, rightWayMergeHashMap(snapshotEdge, newEdge.attributes)))
+    }
+    case (None, Some(newEdge)) => newEdge match {
+      case newEdge.action == DELETE => None
+      case newEdge.action == UPDATE => Some(Edge(x._1._1, x._1._2, newEdge.attributes))
+    }
+  })
+  newSnapshotEdges
+}
+def applyVertexLogs(snapshot: Graph[LTSV.Attributes, LTSV.Attributes], logs: RDD[LogTSV]): RDD[(VertexId, Attributes)] = {
+  val previousSnapshotVertices = snapshot.vertices.map(vertex => (vertex._1, vertex._2))
+
+  // Get all ids of vertices in the log interval
+  val vertexIds = logs.flatMap(getVertexIds)
+
+  // Squash all actions on each vertex to one single action
+  val verticesWithAction = vertexIds.map(id => (id, logs.filter(log => log.objectType match {
+    case VERTEX(objId) => id == objId
+    case EDGE(_, _) => false
+  })))
+    .map(vertexWithActions => (vertexWithActions._1, mergeLogTSVs(vertexWithActions._2)))
+
+  // Combine vertices from the snapshot with the current log interval
+  val joinedVertices = previousSnapshotVertices.fullOuterJoin(verticesWithAction)
+
+  // Create, update or omit (delete) vertices based on values are present or not.
+  val newSnapshotVertices = joinedVertices.flatMap(x => x._2 match {
+    case (Some(snapshotVertex), None) => Some((x._1, snapshotVertex))
+    case (Some(snapshotVertex), Some(newVertex)) => newVertex match {
+      case newVertex.action == DELETE => None
+      case newVertex.action == UPDATE => Some((x._1, rightWayMergeHashMap(snapshotVertex, newVertex.attributes)))
+    }
+    case (None, Some(newVertex)) => newVertex match {
+      case newVertex.action == DELETE => None
+      case newVertex.action == UPDATE => Some((x._1, newVertex.attributes))
```

Comment:
Her mangler også. Create. I fix in the morning. Hodet mitt var ikke helt på plass i dag, så fikk ikke gjort så mye